src/each-repository.ts

```diff
@@ -11,21 +15,42 @@ export function eachRepositoryFactory(app: App) {
 
 export async function eachRepository(
   app: App,
-  callback: EachRepositoryFunction
+  queryOrCallback: EachRepositoryQuery | EachRepositoryFunction,
+  callback?: EachRepositoryFunction
```

Comment:
Do you think is possible to try to overload the signature of this function https://www.typescriptlang.org/docs/handbook/functions.html#overloads ?

something like this:

```typescript
type EachRepositoryQuery = { query: EachRepositoryQuery, callback?: EachRepositoryFunction }

export async function eachRepository(app: App, options: EachRepositoryFunction): Promise<void>
export async function eachRepository(app: App, options: EachRepositoryQuery): Promise<void>

export async function eachRepository(
  app: App,
  queryOrCallback: EachRepositoryQuery | EachRepositoryFunction
) {
...
```

Reply:
Thank you!

The method signature is actually already being overloaded when it is bound onto `app.eachRepository` (which has type `EachRepositoryInterface`, updated in src/types.ts:49), so if one tries to use `app.eachRepository` they will see the two signatures instead of the confusing `queryOrCallback` implementation (vscode screenshot):

<img width="671" alt="Screen Shot 2021-01-08 at 09 10 53" src="https://user-images.githubusercontent.com/3820/104014283-8ea1ac80-5191-11eb-97fb-614e67d3e032.png">

I believe your suggestion of using one `options` parameter would change usage from:

```ts
await app.eachRepository({ installationId }, ({ octokit, repository }) => /* ... */)
```

... to something more verbose:

```ts
await app.eachRepository({ query: { installationId }, callback: ({ octokit, repository }) => /* ... */ })
```

I like the existing version better, but if you guys think that's preferred I can update it!

Reply:
Nice! The hover of VSCode looks perfect to me :)
I thought overload was not entering in action here so you were able to call with double callback or something like that but not. Looks good to me :)

Let's wait for @gr2m double check here ;)